plugins/fluent-bit/integration/tests/tests.bats

```diff
@@ -20,7 +20,11 @@ setup() {
 }
 
 @test "read stdin via fluent-bit" {
-  echo '{"foo": {"bar": 42}}' | check tenzir 'fluent-bit stdin | drop timestamp'
+  # fluent-bit has an internal race condition that can cause the loss of
+  # events during startup. We have to accept this for now and avoid
+  # test flakyness with a bit of trickery.
+  { echo '{"foo": {"bar": 42}}'; sleep 1; echo '{"foo": {"bar": 42}}'; } \
+    | check tenzir 'fluent-bit stdin | drop timestamp | tail 1'
```

Comment:
Given your explanation of the race condition, shouldn't this suffice?

```suggestion
  { sleep 1; echo '{"foo": {"bar": 42}}'; } \
    | check tenzir 'fluent-bit stdin | drop timestamp'
```

Reply:
I don't know. It depends on whether the init logic is triggered on startup or when the first input arrives.

Reply:
The engine is created before the first yield, i.e., as part of the operator's startup.

Reply:
I was referring to what happens in fluent-bit internally.
In any case, I observed the problem once now even with the change as is.

Reply:
I updated the test once more. It doesn't look as clean as in the beginning, but it was to only way to make it reliable without adding redirects directly to the binary.